File: examples/java/core/wait/strategy-helper.java
import java.time.Duration;
import java.util.Map;
import software.amazon.lambda.durable.retry.JitterStrategy;
import software.amazon.lambda.durable.retry.WaitStrategies;

var defaultStrategy = WaitStrategies.<Map<String, String>>defaultStrategy();

var exponentialStrategy = WaitStrategies.<Map<String, String>>exponentialBackoff(
        10,                          // maxAttempts
        Duration.ofSeconds(5),       // initialDelay
        Duration.ofMinutes(5),       // maxDelay
        2.0,                         // backoffRate
        JitterStrategy.FULL);        // jitter

var fixedDelayStrategy = WaitStrategies.<Map<String, String>>fixedDelay(
        10,                          // maxAttempts
        Duration.ofSeconds(30));     // fixedDelay

File: examples/csharp/core/wait/strategy-helper.cs
using Amazon.Lambda.DurableExecution;

public record JobState(string Status);

public static class StrategyHelper
{
    // Build an exponential-backoff strategy from common parameters. The
    // isDone predicate stops polling once the latest state satisfies it.
    public static IWaitStrategy<JobState> Create() =>
        WaitStrategy.Exponential<JobState>(
            maxAttempts: 10,
            initialDelay: TimeSpan.FromSeconds(5),
            maxDelay: TimeSpan.FromMinutes(5),
            backoffRate: 2.0,
            jitter: JitterStrategy.Full,
            isDone: state => state.Status == "COMPLETED");
}
